plugin/flags_sources/compilation_db.py

```diff
@@ -92,10 +92,20 @@ def get_flags(self, file_path=None, search_scope=None):
         if not db:
             log.debug("[db]: not found, return None.")
             return None
+        # First, check if the file is known to the DB:
         if file_path and file_path in db:
             self._cache[file_path] = current_db_path
             File.update_mod_time(current_db_path)
             return db[file_path]
+        # Next, use the header to source map to try to find a related source
+        # file:
+        related_file_path = self._find_related_sources(file_path, db)
+        if related_file_path is not None:
```

Comment:
If I understand correctly, only one path is possible here. We can either go with `file_path` or with the `related_file_path`, right? We could just pick the one that we will use and call the code below once avoiding duplicating lines 95-98.

Reply:
Fixed in latest version of the branch ;-)